contracts/PoaOperator.sol

```diff
@@ -236,4 +237,107 @@ contract PoaOperator is Adminable {
   function submitPeriod(uint256 _slotId, bytes32 _prevHash, bytes32 _blocksRoot) public {
     _submitPeriod(_slotId, _prevHash, _blocksRoot, 0);
   }
+
+  struct Challenge {
+    address payable challenger;
+    uint32 endTime;
+    address slotSigner;
+  }
+
+  mapping(bytes32 => mapping(uint256 => Challenge)) challenges;
+
+  function getChallenge(bytes32 _period, uint256 _slotId) public view returns (address, uint256, address) {
+    Challenge memory chal = challenges[_period][_slotId];
+    return (chal.challenger, chal.endTime, chal.slotSigner);
+  }
+
+  uint256 chalDuration;
+  uint256 constant CHAL_STAKE = 100000000000000000; // 0.1 ETH
```

Comment:
not sure we can initialize it like that because of Zeppellin OS. See https://docs.openzeppelin.com/sdk/2.5/writing-contracts#avoid-initial-values-in-field-declarations

Let's make it a parameter to `initialize` + setter ?

Reply:
very good point. i should probably take it from exitHandler, and save the hassle here.

Reply:
@troggy i think i addressed all